### ISSUE TITLE
2.0.0 System requirements

### ISSUE THREAD

**Issue by barjac** [drowe67/freedv-gui#764]
I have had a report from a Windows 10 user which may be useful in helping to specify minimum system requirements for RADE - or may even be a bug.
He is using Win 10 and has RADE working OK on this machine:

![image](https://github.com/user-attachments/assets/29864e77-112b-4ecd-9837-d3e7c471156e)

However this machine launches into the GUI fine but crashes on clicking Start.

![image](https://github.com/user-attachments/assets/330a53f4-6c0b-42b1-b2d2-e5d03a26f122)

Sightly off topic I have installed 2.0.0 on a RPi4b with 8GB ram and it never crashes but does not decode anything legible in RADE mode.

**Comment by Tyrbiter** [comment]
I think that the RPi system issue is well ahead of the development situation at present, but if you can identify what is going wrong using console output error messages you may be able to correct the problem.

The Windows report is useful but doesn't mention how much RAM is in the system. A 3rd generation i3 (10 years old?) is likely to be struggling so anything that can be done to help it out would be sensible.

For now development is concentrating on Windows as it gets the best coverage for the effort expended, however it's not the time to concentrate on performance using less capable hardware as the RADE code is not yet implemented in C.

**Comment by tmiw** [comment]
I mean, closing without any meaningful feedback as to why isn't helpful either, so that should be fixed. Definitely needs more investigation in any case.

**Comment by Tyrbiter** [comment]
If the i3 system can have additional RAM added it will give a new data point, but some older systems may not have the ability to add more. Feedback on this would be useful to tell us why this system is crashing.

**Comment by tmiw** [comment]
I was looking at the RADE code and I think I can at least make it not quit on error so we can get a better idea of what's going on. @drowe67, can I go ahead and open a PR to make that change?

**Comment by drowe67** [comment]
@tmiw not now - this is precisely the sort of rabbit holes we need to avoid right now.  Lets stay focused on maturing RADE V1 for modern Desktop machines.  I'm OK to add to https://github.com/drowe67/radae/issues/28 for PLT level consideration in the next phase of development.

**Comment by tmiw** [comment]
> @tmiw not now - this is precisely the sort of rabbit hole we need to avoid right now. Lets stay focused on maturing RADE V1 for modern Desktop machines. I'm OK to add to [drowe67/radae#28](https://github.com/drowe67/radae/issues/28) for PLT level consideration in the next phase of development.

I should clarify. Currently the RADE API just calls `exit()` on error which on Windows doesn't provide sufficient time for errors to be reported (even if only via the console you can enable via Tools->Options->Debugging). The absolute minimum change would be to flush stdout/stderr and add a, say, 5-10 second sleep before exiting to allow screenshots of that particular console window to be taken. By doing so, we'd properly be able to triage this particular issue (especially since this is at least the second report I've gotten of RADE causing freedv-gui to close on startup for Windows users). Otherwise, the alternative is that I get on a call and walk someone having this issue through installing a debugger and attaching to freedv-gui to get a stack trace or other actionable output, which would suck up more time IMO.

Of course, we really should improve error handling in the long run, but I'll go ahead and add that to the v2.0 list you linked above.

**Comment by Roturbo** [comment]
Hi,

Some friend have inform me, and just for reference the actual FreeDV 
works on low machines as Dual Core E6750 @ 2.66Ghz

Desktop
Processor Intel(R) Core(TM)2 Duo CPU  E6750  @ 2.66GHz 
RAM       3,50 GB (3,36 GB usable)
win10 Pro  64 bits
_________________________
Laptop
Processor    AMD Athlon Silver 3050U with Radeon Graphics      2.30 GHz
RAM installed	    4,00 GB (3,38 GB usable)
win10 home 64 bits

**Comment by tmiw** [comment]
> > If the i3 system can have additional RAM added it will give a new data point, but some older systems may not have the ability to add more. Feedback on this would be useful to tell us why this system is crashing.
> 
> The working system (Sandy Bridge) has 16GB and the non-working (Ivy Bridge) has 8GB. Note that these are both i3 64bit systems.

FWIW FreeDV only seems to be using a few hundred MB of RAM on my machine when set to RADE (just listening, no signal on frequency). I would think that'd be well within the capabilities of an 8 GB machine, at least immediately on startup.

**Comment by barjac** [comment]
He has tested with AV disabled and no other large applications like browsers running, with no change.

**Comment by tmiw** [comment]
> He has tested with AV disabled and no other large applications like browsers running, with no change.

I can't really do much to debug this right now (especially since the long term plan is to port RADE to C) but I can tell you a workaround I had to do in a Windows VM that I was using which may help. Basically, open Registry Editor and make sure `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Shell Folders` has the following keys:

* AppData
* Common AppData
* Local AppData

If one of these is missing, add it as a "string" key with a path that makes sense (i.e. if "Common AppData" has `C:\Users\User\AppData\Roaming`, make sure "AppData" also has that value). You'll likely need to either reinstall the preview version of FreeDV after doing this, or you can try rerunning `C:\Program Files\FreeDV-2.0.0-devel-2024-10-18-b6d65bc2\bin\rade-setup.bat` as administrator.

**Comment by barjac** [comment]
I have pointed two users to the above and both are either afraid of touching the registry or just too confused by it. Understandably.

I am currently using a LIVE (persistent) Mageia 9 linux system on a USB stick with freedv-2.0 installed and it is performing perfectly on air, so I am considering providing an image for this on request for people with this Windows issue to test/use.

The gz compressed image is currently around 9GB for a plasma install and would fit on a 32GB stick (currently on a 64GB for testing), but this could be reduced with a lightweight D.E.

WIP... :)

**Comment by tmiw** [comment]
I think the Windows issue should still be solved in some manner, so hopefully those willing to use `regedit.exe` will be able to provide some feedback soon (or we get to the point where Python isn't needed anymore, making this moot).

BTW RADE seems to work on the Raspberry Pi 5 using automated tests but I can't comment on audio quality, so this is definitely not an endorsement.

**Comment by Tyrbiter** [comment]
I had a quick look at this, it shows up as the MS Visual C++ redistributable 2015-2022 x86_64 package, it's already installed on my Windows 10 machine which is probably why I have not see this problem with the RADE freedv-gui.

**Comment by barjac** [comment]
> @barjac, I was testing on another Windows 10 machine and it turned out to have the issue you were describing. I was able to fix it by installing the VS2019 redistributable package: https://aka.ms/vs/17/release/vc_redist.x64.exe. Try suggesting that to the people you've heard issues from and see if that helps? If so, I might be able to include it in the next installer.

So far 50% success from 2 cases, one more yet to report back. :)

**Comment by barjac** [comment]
> I tested again in a Pi4B today using a recording of a RADE signal on the system to avoid the load of a browser. It decodes fine but only 50% of the audio i.e. about 1 sec on and 1 second off. I can recognise the person's voice and the quality of what is decoded is good. With a browser running it gets much worse. I have used the Pi as a remote SDR monitor but no longer possible - shame.

Further to this I have now tested with ms-rade-cport on the RPi4 and its worse than before. Only very short bursts of unintelligible decode. Hundreds of lines of WARN...from main.cpp:3094: RX FIFO full.
These are interspersed with occasional single lines of the usual state:....

**Comment by barjac** [comment]
> @barjac, I was testing on another Windows 10 machine and it turned out to have the issue you were describing. I was able to fix it by installing the VS2019 redistributable package: https://aka.ms/vs/17/release/vc_redist.x64.exe. Try suggesting that to the people you've heard issues from and see if that helps? If so, I might be able to include it in the next installer.

Not sure if you have included this in the Win installer but it has fixed several cases recently - one today in fact.

**Comment by tmiw** [comment]
> > @barjac, I was testing on another Windows 10 machine and it turned out to have the issue you were describing. I was able to fix it by installing the VS2019 redistributable package: https://aka.ms/vs/17/release/vc_redist.x64.exe. Try suggesting that to the people you've heard issues from and see if that helps? If so, I might be able to include it in the next installer.
> 
> Not sure if you have included this in the Win installer but it has fixed several cases recently - one today in fact.

Yep, it's in the Windows installer now. 👍

**Comment by mcalistri** [comment]
Just for your information, the FIFO error I faced on freedv-rade, is never appearing using latest freedv-gui: FreeDV GUI 1.9.10-devel.

**Comment by tmiw** [comment]
> Just for your information, the FIFO error I faced on freedv-rade, is never appearing using latest freedv-gui: FreeDV GUI 1.9.10-devel.

Yeah, that logging message is new for 2.0. It still tracks FIFO empty/full events in Tools->Options->Debugging, however, so it's possible that some systems were actually running into this condition and not realizing it.

**Comment by tmiw** [comment]
Closing this out as https://freedv.org/radio-autoencoder/ covers system requirements.
